### DISCUSSION TITLE
App throws CORS error only when navigation occurs

### DISCUSSION THREAD

**Discussion by r25s** [nuxt/nuxt#20697] [Questions] [answered]
I have a simple app, where with my home page, `index.vue`, as below:

```
<template>
  <h1>{{global.data?.title}}</h1>
  <p>{{global.data?.description}}</p>
</template>

<script setup lang="ts">
import {useAsyncData} from "#app";
import { ref } from "vue";

const { $directus } = useNuxtApp()
const { data: global } = await useAsyncData('global', () => {
  return $directus.items('global').readByQuery()
})
</script>
```

It is fetching data from Directus and showing it in the template. It works fine when you load the page directly in via the address bar, or when you refresh the page. But when you navigate to the page via `<NuxtLink to="/">`, the navigation doesn't complete (it stays on the page being navigated from) and a CORS error is shown in the Network tab in Dev Tools. 

The console also shows the following error: `GET ://localhost:8055/items/global net::ERR_FAILED`

I have tried using the nuxt-security module to allow CORS from all origins, methods, and headers. However this has no effect - not surprisingly because I don't actually believe there is a CORS issue or the data fetching would fail all the time and not just on navigation.

Any ideas why this is happening? Could it be a bug?

Thanks!

**Comment by warflash** [comment, marked as answer]
The data fetching doesn't fail on initial load as the server doesn't care about cors, just the browser/client does. Make sure the responses you're getting from directus have the right headers to allow all origins, that should already fix your issue.

**Reply by warflash** [reply]
@ron-wade Did that help solve your issue or are you still stuck on this?

**Reply by r25s** [reply]
Yes, it helped. Turns out I didn't properly set the header in Directus the first time I tried.

Thanks.
